src/mbgl/sprite/sprite_atlas.cpp

```diff
@@ -189,11 +189,10 @@ void SpriteAtlas::updateDirty() {
             holder.texture = spriteIterator->second;
             if (holder.texture != nullptr) {
                 copy(holder, imageIterator->first.second);
+                ++imageIterator;
             } else {
-                images.erase(imageIterator);
+                images.erase(imageIterator++);
```

Comment:
And of course, after writing that I realized I had it backwards. The current code is correct!